### PR TITLE
Fixes/#255 finish e gon100 re hydrogen bus etrago

### LINKED ISSUE #255: finish-eGon100RE-HydrogenBusEtrago
The dataset `HydrogenBusEtrago` is failing when generating the eGon100RE scenario. Related to deprecated function `append`

### PR DESCRIPTION
**Pull request by CarlosEpia**
Fixes #255.

### CODE REVIEW

File: src/egon/data/datasets/scenario_parameters/parameters.py [hunk 1]
```diff
@@ -797,6 +798,7 @@ def gas(scenario):
             "CH4_to_H2": read_costs(costs, "SMR", "efficiency"),
             "H2_to_CH4": read_costs(costs, "methanation", "efficiency"),
             "OCGT": read_costs(costs, "OCGT", "efficiency"),
+            "H2_feedin": 1,
```
Comment (ClaraBuettner) [inline src/egon/data/datasets/scenario_parameters/parameters.py]: We dropped the H2 feedin in the end of eGon. Is it still part of egon-data/powerd-data?

File: src/egon/data/datasets/hydrogen_etrago/bus.py [hunk 2]
```diff
@@ -35,26 +35,33 @@ def insert_hydrogen_buses(scenario="eGon2035"):
...
-    hydrogen_buses = initialise_bus_insertion(
-        carrier, target, scenario=scenario
-    )
-    insert_H2_buses_from_CH4_grid(hydrogen_buses, carrier, target, scenario)
+    s = config.settings()["egon-data"]["--scenarios"]
+    scn = []
+    if "eGon2035" in s:
+        scn.append("eGon2035")
+    if "eGon100RE" in s:
+        scn.append("eGon100RE")
```
Comment (ClaraBuettner) [inline src/egon/data/datasets/hydrogen_etrago/bus.py:46]: I can somehow not find where this variable "scn" is used later on.

File: src/egon/data/datasets/gas_neighbours/eGon100RE.py [hunk 3]
```diff
@@ -111,9 +111,32 @@ def define_DE_crossbording_pipes_geom_eGon100RE(scn_name="eGon100RE"):
         geometry (geom and topo) but no capacity.
 
     """
+
+    def find_equivalent_H2(df):
+        equivalent = {}
+        H2 = df[df["carrier"].isin(["H2", "H2_grid"])].set_index("bus_id")
```
Comment (ClaraBuettner) [inline src/egon/data/datasets/gas_neighbours/eGon100RE.py]: Are both carriers really needed here? I somehow thought we only have one H2 grid carrier

**Comment by ClaraBuettner** [comment]
There are a lot of changes in this PR and I am not sure if I really checked it in a good way. But I guess it will work out like this. I made some comments in the code.

**Comment by ClaraBuettner** [comment]
Did you had a look at the (still open) PR in egon-data when implementing these changes?

File: src/egon/data/datasets/scenario_parameters/parameters.py (hunk 1 above)
Comment (CarlosEpia) [reply src/egon/data/datasets/scenario_parameters/parameters.py]: Right. H2_feedin is used in for eGon2035 [here](https://github.com/openego/eGon-data/blob/7cf077b950edbb4356aac00f770a93510fa17c94/src/egon/data/datasets/hydrogen_etrago/h2_to_ch4.py#L116-L139), but those links are not copied for eGon100RE [here](https://github.com/openego/eGon-data/blob/7cf077b950edbb4356aac00f770a93510fa17c94/src/egon/data/datasets/hydrogen_etrago/h2_to_ch4.py#L142-L146).

Comment (CarlosEpia) [reply src/egon/data/datasets/scenario_parameters/parameters.py]: Changes to discard H2_feedin for eGon100RE applied in af66d9498aa27e4aafb6b6f7b5be07396880563d and 3555507f8a3e3dbd31f42c989db211502edff19b

File: src/egon/data/datasets/hydrogen_etrago/bus.py (hunk 2 above)
Comment (CarlosEpia) [reply src/egon/data/datasets/hydrogen_etrago/bus.py:46]: Fixed in 9b254724f35f014ec7074057448ad2ce5ff7d3d6

File: src/egon/data/datasets/gas_neighbours/eGon100RE.py (hunk 3 above)
Comment (CarlosEpia) [reply src/egon/data/datasets/gas_neighbours/eGon100RE.py]: The hydrogen buses in Germany have carrier _H2_grid_, but the foreign buses have carrier _H2_. As we talked, it makes sense to unify the carrier names. After looking over the two options, I think we should unify with the name _H2_. The required changes are not that extensive but the consistency of the model will improve significantly.

Comment (ClaraBuettner) [reply src/egon/data/datasets/gas_neighbours/eGon100RE.py]: Did you also check if changes in eTraGo are needed?

Comment (CarlosEpia) [reply src/egon/data/datasets/gas_neighbours/eGon100RE.py]: Yes. Some changes are required.

**Comment by CarlosEpia** [comment]
This branch was tested successfully for Germany.

**Review by ClaraBuettner — APPROVED** [review]
When we make sure that eTraGo can be compatible to the new and old names for hydrogen carriers, I agree with theses changes

**Pull request merged**